plugins/coverbrowser.koplugin/listmenu.lua

```diff
@@ -743,19 +785,16 @@ function ListMenu:_recalculateDimen()
         self.itemnum_orig = self.path_items[self.path]
         self.focused_path_orig = self.focused_path
     end
-    local available_height = self.dimen.h - self.others_height
+    local available_height = self.dimen.h - self.others_height - Size.line.thin
 
-    -- 64 hardcoded for now, gives 10 items both in filemanager
-    -- and history on kobo glo hd
-    local item_height_min = Screen:scaleBySize(64)
-    self.perpage = math.floor(available_height / item_height_min)
+    self.perpage = BookInfoManager:getSetting("files_per_page") or math.floor(available_height / Screen:scaleBySize(64))
     self.page_num = math.ceil(#self.item_table / self.perpage)
     -- fix current page if out of range
     if self.page_num > 0 and self.page > self.page_num then self.page = self.page_num end
 
-    local height_remaining = available_height - self.perpage * item_height_min
-    height_remaining = height_remaining - (self.perpage+1) -- N+1 LineWidget separators
-    self.item_height = item_height_min + math.floor(height_remaining / self.perpage)
+    -- menu item height based on number of items per page
+    -- add space for the separator
+    self.item_height = math.floor(available_height / self.perpage) - Size.line.thin
```

Comment:
You could compute all the font sizes here, to avoid recomputing it for each item (but that would may be make the whole thing less readable, not seeing the font size near the font name), so all that magic is contained and explained in `ListMenu:_recalculateDimen()`.
```lua
fontsize_directory_nb_items = math.min(20, _fontSize(15)))
fontsize_direction_name = fontSize(20))
fontsize_fileinfo = math.min(20, _fontSize(14)) -- don't go too large to give most room to title & authors
fontsize_title = _fontSize(20)
fontsize_authors = _fontSize(18)
fontsize_noinfo = _fontSize(18) -- you forgot that one
```
The last one is another you need to update, in the `else -- bookinfo not found`, it's the one displayed with the filename and an added "..." while waiting for the cover to be fetched. It's currently too large when large number of items.

Anyway, I'm fine with the look now, but you're probably not :)
So, if you want to introduce another setting, `Zoom font : -2 -1 0(default) 1 2 3 4 5 6` (or "Font size shift", dunno) , your `_fontSize()` would use that saved setting to tweak the formula.
Probably one of these 2 solutions (not sure which is best, probably the 1st for consistency).
```lua
local font_size = math.ceil( (nominal+font_zoom) * dimen.h / 64)
local font_size = math.ceil(nominal * dimen.h / 64) + font_zoom
```

Reply:
I'll check if this function triggers when files_per_page changes. If yes, then I'll go with your suggestion. The extra `zoom font` setting may not be necessary.

Reply:
Decided not to move font calculation to `ListMenu:_recalculateDimen()` because of its dependency on `ListMenuItem.underline_h`. I feel that apart from limited readability, it can introduce hard-to-track visual glitches when someone changes `ListMenuItem`.